src/main/java/com/minecrafttas/tasmod/TASmodClient.java

```diff
@@ -0,0 +1,177 @@
+package com.minecrafttas.tasmod;
+
+import java.io.File;
+import java.io.IOException;
+
+import com.minecrafttas.common.Configuration;
+import com.minecrafttas.common.Configuration.ConfigOptions;
+import com.minecrafttas.common.KeybindRegistry;
+import com.minecrafttas.common.events.EventListener;
+import com.minecrafttas.common.events.client.EventClientInit;
+import com.minecrafttas.common.events.client.player.EventPlayerJoinedClientSide;
+import com.minecrafttas.common.events.client.player.EventPlayerLeaveClientSide;
+import com.minecrafttas.tasmod.gui.InfoHud;
+import com.minecrafttas.tasmod.handlers.InterpolationHandler;
+import com.minecrafttas.tasmod.handlers.KeybindingHandler;
+import com.minecrafttas.tasmod.handlers.LoadingScreenHandler;
+import com.minecrafttas.tasmod.networking.TASmodNetworkClient;
+import com.minecrafttas.tasmod.playback.PlaybackSerialiser;
+import com.minecrafttas.tasmod.playback.server.InitialSyncStatePacket;
+import com.minecrafttas.tasmod.tickratechanger.TickrateChangerClient;
+import com.minecrafttas.tasmod.util.ShieldDownloader;
+import com.minecrafttas.tasmod.util.TickScheduler;
+import com.minecrafttas.tasmod.virtual.VirtualInput;
+import com.minecrafttas.tasmod.virtual.VirtualKeybindings;
+
+import net.fabricmc.api.ClientModInitializer;
+import net.fabricmc.loader.impl.FabricLoaderImpl;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.entity.EntityPlayerSP;
+
+public class TASmodClient implements ClientModInitializer, EventClientInit, EventPlayerJoinedClientSide, EventPlayerLeaveClientSide{
+
+
+	public static boolean isDevEnvironment;
+
+	public static VirtualInput virtual;
+
+	public static PlaybackSerialiser serialiser = new PlaybackSerialiser();
+
+	public static final String tasdirectory = Minecraft.getMinecraft().mcDataDir.getAbsolutePath() + File.separator + "saves" + File.separator + "tasfiles";
+
+	public static final String savestatedirectory = Minecraft.getMinecraft().mcDataDir.getAbsolutePath() + File.separator + "saves" + File.separator + "savestates";
+
+	public static InfoHud hud;
+	
+	public static ShieldDownloader shieldDownloader;
+	
+	public static TASmodNetworkClient packetClient;
+	
+	public static TickrateChangerClient tickratechanger = new TickrateChangerClient();
+	
+	public static TickScheduler gameLoopSchedulerClient = new TickScheduler();
+	
+	public static TickScheduler tickSchedulerClient = new TickScheduler();
+	
+	public static Configuration config;
+	
+	public static LoadingScreenHandler loadingScreenHandler;
+	
+	public static KeybindingHandler keybindingHandler;
+	
+	public static InterpolationHandler interpolation = new InterpolationHandler();
```

Comment:
please don't put spaces everywhere and add a small // comment after the class explaining what it does if it's not entirely obvious :)

Reply:
Will be fixed in my upcoming Documentation🚂 PR, after I'm done with fully migrating everything... The spaces are there to remind me to add javadocs